### PR TITLE
scripts(lint-packages): add indentation checks

### PR DESCRIPTION
**Pull request by TomJo2000**
Alright.
Finally finished writing the indent linter.

This will flag any instances of space for indentation,
or mixed tabs and spaces.

With specific exceptions for array alignment (spaces)
and Heredocs (both spaces and mixed indentation permitted due to whatever the hell is going on in [`ghc-libs`](https://github.com/termux/termux-packages/blob/a08e4aa1a43816db5690a53ebea218206cea4edb/packages/ghc-libs/build.sh#L78-L109))

On top of this we have ***109*** packages that will need indentation fixes to comply with the coding guidelines this linter addition is now enforcing.
I will be pushing the necessary changes with `[no ci]`, broken up into batches shortly.
Since this is just whitespace changes to the effected build files, a mass rebuild will not be necessary.

This addition turned out more complicated than anticipated, but everything is working well, and all package pass linting in my local testing.
Any questions or comments are appreciated.

### CODE REVIEW

**Comment by TomJo2000** [comment]
Well I guess that's what happens to the reviews when 109 packages need whitespace fixes...

**Review by ravener — APPROVED** [review]

File: scripts/lint-packages.sh [hunk 1]
```diff
@@ -66,6 +66,52 @@ check_package_name() {
...
+		}
+
+		[[ "$line" == *'=('* ]] && in_array=1
+		if (( ! in_array )); then # spaces for indentation are okay for aligning arrays
+			grep -qP '^ +' <<< "$line" && { # check for spaces as indentation
+				issues[1]='Use tabs for indentation'
+				bad_lines[$i]="${pkg_script}:${i}:$line"
+			}
+		fi
+		[[ "$line" == *')'   ]] && in_array=0
```
Comment (truboxl) [inline scripts/lint-packages.sh:103]:
```suggestion
		[[ "$line" == *')' ]] && in_array=0
```

File: scripts/lint-packages.sh [hunk 2]
```diff
@@ -98,6 +144,15 @@ lint_package() {
...
 	echo "PASS"
 
+	echo -n "Indentation check: "
+	for script in "$package_script" $(dirname "$package_script")/*.subpackage.sh; do
+		test ! -f "$script" && continue
+		check_indentation "$script" || return 1
+	done
+	echo "PASS"
+
+	echo
```
Comment (truboxl) [inline scripts/lint-packages.sh:154]: Extra echo needed?

File: scripts/lint-packages.sh (hunk 1 above)
Comment (TomJo2000) [reply scripts/lint-packages.sh:103]: That was intentional to line it up with `[[ "$line" == *'=('* ]] && in_array=1`, but I forgot to change it to a single space when I moved it around.

File: scripts/lint-packages.sh (hunk 2 above)
Comment (TomJo2000) [reply scripts/lint-packages.sh:154]: Not needed, just looks better.
I can make it a `echo -e "PASS\n"` above instead.

Comment (TomJo2000) [reply scripts/lint-packages.sh:154]: Same thing is done at the trailing whitespace check actually.

File: scripts/lint-packages.sh [hunk 3]
```diff
@@ -98,6 +144,15 @@ lint_package() {
 	fi
 	echo "PASS"
 
+	echo -n "Indentation check: "
+	for script in "$package_script" $(dirname "$package_script")/*.subpackage.sh; do
```
Comment (truboxl) [inline scripts/lint-packages.sh:148]: Do we need to apply other checks on subpackage.sh too?

File: scripts/lint-packages.sh (hunk 2 above)
Comment (TomJo2000) [reply scripts/lint-packages.sh:154]: I'll group these first five checks then.
image

Comment (TomJo2000) [reply scripts/lint-packages.sh:154]: image
There we go.

**Review by licy183 — APPROVED** [review]

**Comment by TomJo2000** [comment]
Might wanna hold up on that approval, I have some general housekeeping that I'm finishing up for commit now.

**Comment by TomJo2000** [comment]
Hit a new edgecase with herestrings `<<<` being misidentified as Heredocs `<<`.
Also had some other stuff come up.
Doing final tests now.

**Comment by TomJo2000** [comment]
Dealt with the `herestring` edge case, that's in [`aa79aa0`](https://github.com/termux/termux-packages/pull/19896/commits/aa79aa000e2840acf2ef388d2dfe7cf58ad6d626).

The general cleanup is a separate commit, [`e9dd3f5`](https://github.com/termux/termux-packages/pull/19896/commits/e9dd3f50d1676d7f73dc064f966bbe14765b32d2).

Also found two more packages that had indentation issues, bringing the sum total to 111.

**Comment by TomJo2000** [comment]
Ah shit I didn't push that with no CI...

**Comment by TomJo2000** [comment]
Alright, we got updates failing due to the new trailing whitespace check.
https://github.com/termux/termux-packages/actions/runs/8805195705/job/24167255257

I'm gonna merge this PR so we can get that rebuilding, if anyone finds any major bugs I missed in testing we'll deal with it then.
